tiledb/sm/query/query_condition.cc

```diff
@@ -1867,6 +1867,10 @@ Status QueryCondition::apply_dense(
     const uint64_t stride,
     const void* cell_slab_coords,
     uint8_t* result_buffer) {
+  if (!tree_) {
```

Comment:
We could leave this in there... But eventually as talked above, I'd rather not allow empty conditions at all.

Reply:
I looked back at whether this was still needed and it turns out that after preventing the setting of empty query conditions the only code that reached it was tests I added. I've removed the new tests and this check to go back to the status quo.